src/pages/docs/ai-transport/features/token-streaming/message-per-response.mdx

```diff
@@ -0,0 +1,373 @@
+// History items are delivered as complete messages
+// with the most recent messages first.
+for (const msg of historyPage.items.reverse()) {
+  console.log('Complete response:', msg.data);
+  // msg.data contains the full concatenated text
+  displayFullResponse(msg.data);
+}
+
+// Subscribe to new tokens
+await channel.subscribe((msg) => {
+  switch (msg.action) {
+    case 'message.create':
+      // etc
+    case 'message.append':
+      // etc
+    case 'message.update':
+      // etc
+  }
+});
+```
+</Code>
+
+## Headers and metadata
+
+Use the `extras.headers` field to attach metadata to your messages. Headers from the final
+`message.append` replace all previous headers.  Headers are useful for correlating Ably messages
+with external systems.
+
+### Adding metadata to messages
+
+Include metadata in the initial message or any append operation:
+
+<Code>
+```javascript
+ // Initial message with metadata
+const response = await channel.publish({
+  data: 'Hello',
+  extras: {
+    headers: {
+      responseId: 'resp_12345',        // Your database ID
+      modelRequestId: 'req_abc123',    // AI model request ID
+      model: 'claude-3-opus',
+      temperature: '0.7',
+      userId: 'user_789'
+    }
+  }
+});
+
+// Append with updated metadata
+channel.appendMessage(response.serials[0], ' world!', {
+  extras: {
+    headers: {
+      responseId: 'resp_12345',
+      modelRequestId: 'req_abc123',
+      model: 'claude-3-opus',
+      temperature: '0.7',
+      userId: 'user_789',
+      tokensUsed: '150',
+      completionTime: '2.4s'
+    }
+  }
+});
+```
+</Code>
+
+### Metadata best practices
+
+Do not include metadata in the body of an append request. Instead, use the `extras.headers` field to
+keep metadata separate from the message content. This ensures that clients can easily process the
+concatenated response without needing to parse out metadata.
+
+<Code>
+```javascript
+// ✓ GOOD: Metadata in headers
+const response = await channel.publish({
+  data: 'The response text',  // Pure concatenated text
+  extras: {
+    headers: {
+      model: 'gpt-4',
+    }
+  }
+});
+
+// ✗ BAD: Mixing metadata with content
+const response = await channel.publish({
+  data: JSON.stringify({  // Don't do this
+    text: 'The response text',
+    model: 'gpt-4',
+  })
+});
+```
+</Code>
+
+By including metadata in the body of the message, the final concatenated response would contain all
+the metadata from each append, making it difficult to extract the pure response text.
+
+For example, if you appended tokens with metadata in the body, the final message data would look
+like this:
+
+```json
+{
+  "text": "Hello",
+  "model": "gpt-4",
+}{
+  "text": " world",
+  "model": "gpt-4",
+}{
+  "text": "!",
+  "model": "gpt-4",
+}
+```
+
+If you use headers for metadata, and the body only contains the response text, the final message
+data would be simply:
+
+```text
+Hello world!
+```
+
+## Requirements and restrictions
+
+### Channel rule
```

Comment:
I think this should appear earlier in the page, as the user has no indication that they need to configure the rule to use the code examples above until this point. Also the code examples should have a namespace prefix in the channel name